connectors/sources/microsoft_teams.py

```diff
@@ -144,9 +319,14 @@ async def _fetch_token(self):
             client_credential=self.client_secret,
             authority=authority,
         )
-        token_metadata = auth_context.acquire_token_by_username_password(
-            username=self.username, password=self.password, scopes=SCOPE
-        )
+        if is_acquire_for_client:
+            token_metadata = auth_context.acquire_token_for_client(
+                scopes=[GRAPH_ACQUIRE_TOKEN_URL]
+            )
```

Comment:
there are different type of scopes required for `acquire_token_for_client` and `acquire_token_by_username_password`. For `acquire_token_for_client`  we need [".default"] scope to be added and for `acquire_token_by_username_password` : ["User.Read.All", "TeamMember.Read.All", "ChannelMessage.Read.All", "Chat.Read", "Chat.ReadBasic", "Calendars.Read"] must be passed to generate token.. To make it more clear I can pass  [".default"] instead of the complete url. Both will perform the same operation.

Reply:
If you're sure, no need to change it. I just noticed they were different and wanted to confirm why.

Reply:
The scopes are different only. For `acquire_token_for_client` we need to pass `default` in scope and for `acquire_token_by_username_password` we need to pass the scope which is related to fetching chats and teams metadata.
Passing [".default"] or [""https://graph.microsoft.com/.default"] for `acquire_token_for_client` is same thing.